get-git/getter.go

```diff
@@ -20,32 +21,41 @@ import (
 )
 
 func NewGetter(stoic stoic.Stoic, tool stoic.Tool) (tool.Getter, error) {
-	url := tool.Endpoint()
-	branch := Branch(tool.Channel())
-	if !branch.IsValid() {
-		return nil, errors.Errorf("invalid branch name, '%v'", branch)
+	var options Options
+	err := mapstructure.Decode(tool.Config().Getter.Options, &options)
+	if err != nil {
+		return nil, err
 	}
 
-	pathElems := []string{stoic.Root(), "git", url.Hostname()}
-	pathElems = append(pathElems, strings.Split(url.EscapedPath(), "/")...)
-	gitDir := filepath.Join(pathElems...)
+	if options.URL == nil {
+		options.URL = tool.Endpoint()
+	}
+	if options.Branch == "" {
+		options.Branch = Branch(tool.Channel())
+	}
 
-	return &gitGetter{
-		url:    url,
-		branch: branch,
+	if !options.Branch.IsValid() {
+		return nil, errors.Errorf("invalid branch name, '%v'", options.Branch)
+	}
+
+	pathElems := []string{stoic.Root(), "git", options.URL.Hostname()}
+	pathElems = append(pathElems, strings.Split(options.URL.EscapedPath(), "/")...)
+	gitDir := filepath.Join(pathElems...)
 
-		gitDir: gitDir,
-	}, nil
+	return &Getter{options, gitDir}, nil
 }
 
-type gitGetter struct {
-	url    *url.URL
-	branch Branch
+type Options struct {
+	URL    *url.URL
+	Branch Branch
+}
 
+type Getter struct {
+	Options
 	gitDir string
 }
 
-func (gg gitGetter) runNativeGit(command string, args ...string) error {
+func (gg Getter) runNativeGit(command string, args ...string) error {
```

Comment:
Not really a thing, but this double `gg` is killing my OCD

Reply:
You want a single one?

Reply:
I had noticed, but was trying to ignore it.